### ISSUE TITLE
Can't find docs on replying to events

### ISSUE THREAD

**Issue by ahmetb** [knative/docs#2105]
One can reply to Knative events with events.

This is very briefly mentioned at https://knative.dev/docs/eventing/samples/sequence/sequence-reply-to-event-display/
![image](https://user-images.githubusercontent.com/159209/72471083-cbadab80-3796-11ea-980f-bb549a14064b.png)

as well as https://knative.dev/v0.9-docs/eventing/parallel/
![image](https://user-images.githubusercontent.com/159209/72471123-dff1a880-3796-11ea-9de5-df11cb8a565a.png)

However nowhere I can clearly tell

- I can reply to events with events
- how to reply

**Comment by aslom** [comment]
@lionelvillard I think it is related to data-plane defining Callable and also to work on async?

**Comment by vaikas** [comment]
We also need to vet all the places where a user can reply and then put a link to examples on how to reply. We might be able to leverage cloudevents sdk, but maybe the examples in all the languages are not complete.

**Comment by abrennan89** [comment]
@vaikas happy to help with this if there's a list or some spec info somewhere about which resources can have replies specified?

**Comment by vaikas** [comment]
Thanks for the ping @abrennan89 I'll try to get to it either this week or next.

**Comment by lionelvillard** [comment]
seems like it's now documented here: https://github.com/knative/specs/blob/main/specs/eventing/data-plane.md#derived-reply-events

**Comment by abrennan89** [comment]
Thanks @lionelvillard - I think we'll also need some actual examples of YAML though? It's not clear to me _where_ / how a user actually sets these from this doc?

**Comment by lionelvillard** [comment]
I think the original issue is about the event receiver can send back a CloudEvent reply. Maybe I'm reading this wrong. 

Since this is about event delivery, seems like the right place to add a section about reply is here: https://knative.dev/docs/developer/eventing/event-delivery/

WDYT?

**Comment by snneji** [comment]
@abrennan89, I'm not sure what the priority is on this one. Could you add a priority label?

**Comment by lionelvillard** [comment]
This is quite high-priority for GA.

@abrennan89 I agree with should have YAML examples:
- https://knative.dev/docs/eventing/channels/subscriptions/: there is no mention of reply. It should
- https://knative.dev/docs/eventing/broker/: `Subscribers can also reply to an active request from the broker, and can respond with a new CloudEvent that will be sent back into the broker.`. This is good and it should include a link to reply documentation (see below)
- There are also example of reply in Parallel/Sequence, but this is out-of-scope for GA 

The same way there is documentation for Sources, Delivery and Sink, I wonder, should we also have a `reply` section for extra visibility?

**Comment by abrennan89** [comment]
@lionelvillard who can provide these samples though? Can you assign someone please since Ville is likely not to work on this now?

I disagree that this should have its own section, since this opens the door for having much too granular docs where specific sections of the spec have their own section in docs each time, and we're trying to get away from that and group things in a more meaningful way.
My vote would be to add a generic section about replies in the spec that can be included as a snippet for each section like Sources, Delivery, etc, wherever it needs to be.

**Comment by evankanderson** [comment]
#4351 updates the "Event Delivery" section of the Developer Guide, which I think should resolve this issue.

https://knative.dev/docs/eventing/getting-started/ currently explains [sending events to the broker](#https://knative.dev/docs/eventing/getting-started/#sending-events-to-the-broker), but doesnt't have any description of how replies fit into the picture. Were it me, I'd update the TOC as follows:

```
  Creating a Knative Eventing namespace
  Adding a broker to the namespace
  Creating event consumers
  Creating triggers
  Creating a pod as an event producer
  Sending events to the broker
+ Replying to broker events
  Verifying that events were received
  Cleaning up example resources
```

We don't seem to have an equivalent doc for Channels (which seems okay); I think the new file in #4351 should be sufficient for Channels / Subscriptions.
